splink/connected_components.py

```diff
@@ -268,10 +284,57 @@ def _cc_assess_exit_condition(representatives_name):
     return sql
 
 
+def _cc_create_unique_id_cols(linker, match_probability_threshold):
+
+    """
+    Takes the output of linker.predict() and either creates unique IDs for
+    our linked dataframes, if we are performing a link job, or pulls out
+    the unique ID columns if deduping.
+    """
+
+    df_predict = linker.predict()
+
+    # Code assumes a unique ID column is provided.
+    if linker._settings_dict["link_type"] == "dedupe_only":
+        sql = f"""
+            select unique_id_l, unique_id_r
```

Comment:
We should be using:
```
cols = linker._settings_obj._unique_id_input_columns
```
for this because the `unique_id` and `source_dataset` columns may be called somethine else, but it probably requires a little additional refactoring.

Specifically, i think we want a `composite_unique_id.py` that is responsible for these sort of manipulations.

It gets a bit tricky though e.g. [here](https://github.com/moj-analytical-services/splink/blob/dcd241dc5d204b1ac0eb893d7886f3b1eb27769d/splink/blocking.py#L18)

for now, `composite_unique_id.py` probably just wants a single function that takes a list of strings and returns `" || '-__-' || ".join(cols)`